lib/Models/addUserCatalogMember.ts

```diff
@@ -0,0 +1,71 @@
+"use strict";
+
+import Terria from "./Terria";
+import { BaseModel } from "./Model";
+import isDefined from "../Core/isDefined";
+import hasTraits from "./hasTraits";
+import GroupTraits from "../Traits/GroupTraits";
+import when from "terriajs-cesium/Source/ThirdParty/when";
+import TerriaError from "../Core/TerriaError";
+import CommonStrata from "./CommonStrata";
+import Mappable from "./Mappable";
+import defaultValue from "terriajs-cesium/Source/Core/defaultValue";
+import GeoJsonCatalogItem from "./GeoJsonCatalogItem";
+import GroupMixin from "../ModelMixins/GroupMixin";
+
+interface AddUserCatalogMemberOptions {
+    enable?: boolean;
+    open?: boolean;
+    zoomTo?: boolean;
+}
+
+/**
+ * Adds a user's catalog item or group to the catalog.
+ *
+ */
+export default function addUserCatalogMember(
+    terria: Terria,
+    newCatalogMemberOrPromise: BaseModel | Promise<BaseModel | undefined>,
+    optionsArg?: AddUserCatalogMemberOptions
+) {
+    const options: AddUserCatalogMemberOptions = defaultValue(optionsArg, {});
+    return when(newCatalogMemberOrPromise, function(newCatalogItem: BaseModel) {
```

Comment:
Different from how we've done things in the old architecture, we should use Cesium's when.js promises as little as possible. Whenever you get a promise back from Cesium, immediately convert it to a regular promise by calling `makeRealPromise`. Assume that promises everywhere in Terria are real native promises.